### PR TITLE
Adding Back Query Test Coverage

### PR DESCRIPTION
**Pull request by bchong95**
# Adding Back Query Test Coverage

## Description

Adding back single partition and query with continuation tokens to cross partition query test suite.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs [hunk 1]
```diff
@@ -266,23 +351,49 @@ private async Task RunWithApiVersion(string apiVersion, Func<Task> function)
...
+           CollectionTypes collectionTypes,
            IEnumerable<string> documents,
            Query<T> query,
            CosmosClientFactory cosmosClientFactory,
            T testArgs,
            string partitionKey = "/id",
            Cosmos.IndexingPolicy indexingPolicy = null)
         {
-            int retryCount = 1;
+            int retryCount = 5;
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs]: Any rationale for raising retry count?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs [hunk 2]
```diff
@@ -1353,7 +1508,7 @@ private async Task TestQueryCrossPartitionAggregateFunctionsAsync(CosmosContaine
                 foreach (Tuple<string, object> data in datum)
                 {
                     string query = $"SELECT VALUE {data.Item1}(r.{field}) FROM r WHERE r.{partitionKey} = '{uniquePartitionKey}'";
-                    dynamic aggregate = (await QueryWithoutContinuationTokens<dynamic>(
+                    dynamic aggregate = (await RunQuery<dynamic>(
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs]: Qualify with ClassName for static methods.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs [hunk 3]
```diff
@@ -50,6 +50,15 @@ private enum ConnectionModes
             Gateway = 0x2,
         }
 
+        [FlagsAttribute]
+        private enum CollectionTypes
+        {
+            None = 0,
+            NonPartitioned = 0x1,
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs]: NonPartitioned is still not covered right?

**Comment by kirankumarkolli** [comment]
CPTests are differing from master in a big way. FI forward will be harder. Thoughts?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs (hunk 3 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs]: I added the implementation, which doubles the test time, but adds code coverage ( not logic coverage).

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs (hunk 2 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs]: good catch

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs (hunk 1 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.cs]: Yeah I wanted to see if the 500s the emulator gives is sticky or not.

**Comment by j82w** [comment]
> CPTests are differing from master in a big way. FI forward will be harder. Thoughts?

Actually this brings us closer to master. Master was testing all the different container types. I removed it because we blocked the creation of non-partitioned types so there was no easy way to test it.

**Pull request merged**